crates/dbsp/src/storage/backend/posixio_impl.rs

```diff
@@ -80,12 +89,39 @@ impl FileReader for PosixReader {
         }
     }
 
+    fn read_async(
+        &self,
+        blocks: Vec<BlockLocation>,
+        callback: Box<dyn FnOnce(Vec<Result<Arc<FBuf>, StorageError>>) + Send>,
```

Comment:
should this return a future instead?

Reply:
I think that would be a lot harder to work with because it means the caller has to keep polling it, and our callers aren't always running in async contexts.